spec.html

```diff
@@ -42482,104 +42631,222 @@ <h1>
+            1. Return _resultObject_.
+          1. If _t_ is 0 and _mode_ is ~async~, then
+            1. NOTE: There is no special handling of synchronous immediate timeouts. Asynchronous immediate timeouts have special handling in order to fail fast and avoid unnecessary Promise jobs.
+            1. Perform LeaveCriticalSection(_WL_).
+            1. Perform ! CreateDataPropertyOrThrow(_resultObject_, *"async"*, *false*).
+            1. Perform ! CreateDataPropertyOrThrow(_resultObject_, *"value"*, *"timed-out"*).
+            1. Return _resultObject_.
+          1. Let _thisAgent_ be AgentSignifier().
+          1. Let _now_ be the time value (UTC) identifying the current time.
+          1. Let _additionalTimeout_ be an implementation-defined non-negative mathematical value.
+          1. Let _timeoutTime_ be ℝ(_now_) + _t_ + _additionalTimeout_.
+          1. NOTE: When _t_ is +∞, _timeoutTime_ is also +∞.
+          1. Let _waiterRecord_ be a new Waiter Record { [[AgentSignifier]]: _thisAgent_, [[PromiseCapability]]: _promiseCapability_, [[TimeoutTime]]: _timeoutTime_, [[Result]]: *"ok"* }.
+          1. Perform AddWaiter(_WL_, _waiterRecord_).
+          1. If _mode_ is ~sync~, then
+            1. Perform SuspendThisAgent(_WL_, _waiterRecord_).
+          1. Else if _timeoutTime_ is finite, then
+            1. Perform EnqueueAtomicsWaitAsyncTimeoutJob(_WL_, _waiterRecord_).
+          1. Perform LeaveCriticalSection(_WL_).
+          1. If _mode_ is ~sync~, return _waiterRecord_.[[Result]].
+          1. Perform ! CreateDataPropertyOrThrow(_resultObject_, *"async"*, *true*).
+          1. Perform ! CreateDataPropertyOrThrow(_resultObject_, *"value"*, _promiseCapability_.[[Promise]]).
+          1. Return _resultObject_.
+        </emu-alg>
         <emu-note>
-          <p>The embedding may delay notifying _W_, e.g. for resource management reasons, but _W_ must eventually be notified in order to guarantee forward progress.</p>
+          <p>_additionalTimeout_ allows implementations to pad timeouts as necessary, such as for reducing power consumption or coarsening timer resolution to mitigate timing attacks. This value may differ from call to call of DoWait.</p>
         </emu-note>
       </emu-clause>
 
+      <emu-clause id="sec-enqueueatomicswaitasynctimeoutjob" type="abstract operation">
+        <h1>
+          EnqueueAtomicsWaitAsyncTimeoutJob (
+            _WL_: a WaiterList Record,
+            _waiterRecord_: a Waiter Record,
+          ): ~unused~
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Let _timeoutJob_ be a new Job Abstract Closure with no parameters that captures _WL_ and _waiterRecord_ and performs the following steps when called:
+            1. Perform EnterCriticalSection(_WL_).
+            1. If _WL_.[[Waiters]] contains _waiterRecord_, then
+              1. Let _now_ be the time value (UTC) identifying the current time.
```

Comment:
Let's not name two different variables `now` in the same emu-alg.

Reply:
Renamed the one inside the Job closure _timeOfJobExecution_.